File: src/main/java/gg/mineral/practice/util/CoreConnector.java
package gg.mineral.practice.util;

import org.bukkit.Bukkit;

import gg.mineral.practice.PracticePlugin;

public class CoreConnector {
    public static CoreLoader INSTANCE;

    public static boolean connected() {
        return INSTANCE != null;
    }

    static {
        if (Bukkit.getPluginManager().isPluginEnabled("JeezyCore")) {
            INSTANCE = new CoreLoader();

        } else {

            INSTANCE = null;
            PracticePlugin.INSTANCE.getLogger().warning(
                    "The core plugin has failed to link with the practice plugin. Please ensure the core plugin is installed.");
        }
    }

}

File: src/main/kotlin/gg/mineral/practice/util/CoreConnector.kt
package gg.mineral.practice.util

import gg.mineral.practice.PracticePlugin
import org.bukkit.Bukkit

object CoreConnector {
    var INSTANCE: CoreLoader? = null

    fun connected(): Boolean {
        return INSTANCE != null
    }

    init {
        if (Bukkit.getPluginManager().isPluginEnabled("JeezyCore")) {
            INSTANCE = CoreLoader()
        } else {
            INSTANCE = null
            PracticePlugin.INSTANCE.logger.warning(
                "The core plugin has failed to link with the practice plugin. Please ensure the core plugin is installed."
            )
        }
    }
}
